plugins/net_plugin/net_plugin.cpp

```diff
@@ -1499,10 +1499,12 @@ namespace eosio {
             c->enqueue( note );
          }
          c->syncing = true;
-         request_message req;
-         req.req_blocks.mode = catch_up;
-         req.req_trx.mode = none;
-         c->enqueue( req );
+         if( cc.get_block_id_for_num( msg.head_num ) != msg.head_id ) {
```

Comment:
There is a comment block above where we should probably note what this branch "means"

Something like 
```
4.1 My head block number >= peer head block number but they appear to be on a different fork which may be better and should be requested
```

Reply:
At the very least we should note the difference in comments in `develop` I think these backports also should bear the comments though.

Reply:
Good point. I'll update all of them.